01_prepare_host.sh

```diff
@@ -239,14 +239,17 @@ for IMAGE_VAR in $(env | grep -v "_LOCAL_IMAGE=" | grep "_IMAGE=" | grep -o "^[^
   pull_container_image_if_missing "$IMAGE"
  done
 
-if ${IPA_DOWNLOAD_ENABLED}; then
-    # Start image downloader container
-    #shellcheck disable=SC2086
-    sudo "${CONTAINER_RUNTIME}" run -d --net host --name ipa-downloader ${POD_NAME} \
-       -e IPA_BASEURI="$IPA_BASEURI" \
-       -v "$IRONIC_DATA_DIR":/shared "${IPA_DOWNLOADER_IMAGE}" /usr/local/bin/get-resource.sh
-
-    sudo "${CONTAINER_RUNTIME}" wait ipa-downloader
+if ${IPA_DOWNLOAD_ENABLED} || [ ! -f "${IRONIC_DATA_DIR}/html/images/ironic-python-agent.kernel" ]; then
```

Comment:
Not a fan of this if statement, as it executes whatever command is entered in IPA_DOWNLOAD_ENABLED env. It happens to work because true|false are commands too. 

if [ "${IPA_DOWNLOAD_ENABLED}" = true ] makes it a string comparison.

Reply:
Good point, I agree! This would be nice to clean up throughout the repository. Maybe we can do it as a follow up after we have the CI working again

Reply:
It is shell, everything is a string.

Reply:
But I can understand if you like better the the simplified comparison .

Reply:
@Rozzii isn't there a difference between `if "${VARIABLE}"` and `if [[ "${VARIABLE}" == "true" ]]`? I think the first will actually try to execute VARIABLE and check the exit code, whereas the second will do a string comparison.

Reply:
yes, I can see why it can be a security issue

Reply:
@Rozzii If "${VARIABLE}" definitely executes the content of variable. If [ ... does not, as [ is an actual binary in /usr/bin, so the "${VARIABLE}" = true are all just parameters to that command. (disclaimer: [ is built-in for bash, but sh actually calls the [ binary)

Reply:
I have actually missread your comments I thought the original was [[ "${VARIABLE}" == "true" ]] and we are changing it to ${IPA_DOWNLOAD_ENABLED}  but you are doing it the other way around.

So no complaint from me .

Reply:
> yes, I can see why it can be a security issue

This is why I have answered with this comment I thought we are changing with the [[ -less